corehq/apps/locations/models.py

```diff
@@ -274,6 +274,15 @@ def couch_locations(self, wrapped=True):
             return itertools.imap(Location.wrap, locations)
         return locations
 
+    def assigned_to_user(self, domain, user):
+        if user.has_permission(domain, 'access_all_locations'):
+            return self.all()
+
+        assigned_location_ids = user.get_assigned_location_ids(domain)
+        if not assigned_location_ids:
+            return self.none()  # No locations are assigned to this user
+        return self.all() & SQLLocation.active_objects.get_locations_and_children(assigned_location_ids)
+
```

Comment:
@esoergel will this conflict with other uses for `accessible_to_user`. I remember @proteusvacuum added this for his set of changes as well? If this does not then i can surely move this to `accessible_to_user`.